datadog/resource_datadog_authn_mapping.go

```diff
@@ -0,0 +1,203 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+func resourceDatadogAuthnMapping() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog AuthN Mappings resource. This can be used to create and manage Datadog SAML Mappings.",
```

Comment:
Nit: first it's called "Datadog AuthN Mappings", then "Datadog SAML Mappings". 

Could we add a sentence explaining the "who cares?" a bit more? Something like: This feature lets you automatically assign roles to users based on their SAML attributes.